src/abi/mod.rs

```diff
@@ -39,3 +43,16 @@ pub fn generate_abi(
         }
     }
 }
+
+/// Returns a set of all non-unique public function names in a given contract.
+/// These names should not be used in the metadata. Instead, the mangled versions should be used.
+pub fn non_unique_function_names(contract_no: usize, ns: &Namespace) -> HashSet<&String> {
```

Comment:
This function can be `pub(super)`. I guess you don't need to use the `pub` prefix at all.